design-system/src/theme-provider.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  createContext,
+  useLayoutEffect,
+  useState,
+  useMemo,
+  useContext,
+  useCallback,
+  useRef,
+  type ReactNode,
+} from 'react';
+import kebabCase from 'lodash/kebabCase';
+import isEmpty from 'lodash/isEmpty';
+import { warning } from '@commercetools-uikit/utils';
+import { themes, themesNames } from './custom-properties';
+
+type ThemeName = keyof typeof themes;
+
+const allThemesNames = Object.keys(themesNames);
+
+const ThemeContext = createContext({});
+
+const toVars = (obj: Record<string, string>) =>
+  Object.fromEntries(
+    Object.entries(obj).map(([key, value]) => [`--${kebabCase(key)}`, value])
+  );
+
+type ThemeProviderProps = {
+  children: ReactNode;
+  theme?: string;
+};
+
+const validateTheme = (themeName?: string): ThemeName => {
+  if (!themeName) {
+    return 'default';
+  }
+  const isNewThemeValid = allThemesNames.includes(themeName);
+  if (isNewThemeValid) {
+    return themeName as ThemeName;
+  }
+  if (!isNewThemeValid) {
+    console.warn(
```

Comment:
I am not using `warning` function here because it would throw every time someone changes theme in storybook to `Dark theme (experimental)` (and therefore make using strorybook very cumbersome) until other themes are properly defined in `definition.yaml` file